patch_migrations.py

```diff
@@ -0,0 +1,154 @@
+
+blocks_columns = [
+	'hash',
+	'parentHash',
+	'uncleHash',
+	'coinbase',
+	'root',
+	'txRoot',
+	'receiptRoot',
+	'mixDigest',
+	'td',
+	'baseFee',
+	'withdrawalHash',
+]
+
+withdrawals_columns = [
+	'address',
+	'blockHash'
+]
+
+transactions_columns = [
+	'hash',
+	'r',
+	's',
+	'sender',
+	'func'
+]
+
+event_logs_colums = [
+	'address',
+	'transactionHash',
+	# 'transactionIndex', transaction index needs a different fix
+	'blockHash'
+]
+
+columns = {
+    'blocks': [blocks_columns, ('number')],
+    'withdrawals': [withdrawals_columns, ('block', 'wtdrlIndex')],
+    'transactions': [transactions_columns, ('id')],
+    'event_logs': [event_logs_colums, ('block', 'logIndex')],
+}
+
+
+class DatabaseManager:
+    def __init__(self, db_name, table_name):
+        self.db = db_name
+        self.table = table_name
+
+    @contextmanager
+    def connect(self):
+        try:
+            conn = sqlite3.connect(self.db)
+            yield conn
+        except:
+            raise
+        finally:
+            conn.close()
+
+    def execute(self, query):
+        with self.connect() as conn:
+            cursor = conn.cursor()
+            cursor.execute(query, params or ())
+            conn.commit()
+            return cursor.rowcount
+
+    def fetchall(self, query, params=None):
+        with self.connect() as conn:
+            cursor = conn.cursor()
+            cursor.execute(query, params or ())
+            return cursor.fetchall()
+
+    def fetchone(self, query, params=None):
+        with self.connect() as conn:
+            cursor = conn.cursor()
+            cursor.execute(query, params or ())
+            return cursor.fetchone()
+
+class DataManipulator:
+    def __init__(self, connection, table):
+        self.conn = connection
+        self.tbl = table
+        self.prmry = columns.get(table)[1]
+        self.initial = {}
+
+    def get_initial_results(self):
+        primary = self.prmry
+        table = self.tbl
+
+        if isinstance(primary, str):
+            partial = f', {primary} FROM {table};'
+        elif isinstance(primary, tuple):
+            partial = f', {primary[0]}, {primary[1]} FROM {table};'
+
+        for column in columns.get(table)[0]:
+            q = f'SELECT hex({column})' + partial
+            rows = self.conn.fetchall(q)
+            zeros = find_zeros(rows)
+            prepared = remove_zeros(zeros)
+            
+            if len(prepared) > 0:
+                self.initial[column] = prepared
+
+        return self.initial
+
+    def alter_table(self):
+        initial_results = self.get_initial_results()
+
+        primary = self.prmry
+        table = self.tbl
+
+        for k, v in initial_results.items():
+            print(f"processing column: {k} on table: {table} with {len(v)} statements to process")
+            for item in v:
+                if isinstance(primary, str):
+                    s = f'UPDATE {table} SET {k} = {item[0]} WHERE {primary} = {item[1]}'
```

Comment:
If `item[0]` is hex encoded, won't this need to be `SET {K} = X'{item[0]}'`? The values stored in the database should be the raw values, not the hex encoded values.